.github/workflows/ci.yml

```diff
@@ -25,29 +31,44 @@ concurrency:
 jobs:
   setup:
     uses: ./.github/workflows/setup.yml
+    with:
+      linux_amdgpu_families: ${{ inputs.linux_amdgpu_families }}
+      windows_amdgpu_families: ${{ inputs.windows_amdgpu_families }}
 
   build_linux_packages:
     name: Build Linux Packages
     needs: setup
     if: ${{ fromJSON(needs.setup.outputs.enable_build_jobs) }}
+    strategy:
+      matrix:
+        families: ${{ fromJSON(needs.setup.outputs.linux_amdgpu_families) }}
     uses: ./.github/workflows/build_linux_packages.yml
     with:
-      amdgpu_families: "${{ inputs.amdgpu_families != '' && inputs.amdgpu_families || 'gfx94X-dcgpu' }}"
+      amdgpu_families: ${{ matrix.families.target }}
     permissions:
       id-token: write
 
   build_windows_packages:
     name: Build Windows Packages
     needs: setup
     if: ${{ fromJSON(needs.setup.outputs.enable_build_jobs) }}
+    strategy:
+      matrix:
+        families: ${{ fromJSON(needs.setup.outputs.windows_amdgpu_families) }}
     uses: ./.github/workflows/build_windows_packages.yml
+    with:
+      amdgpu_families: ${{ matrix.families.target }}
 
   test_linux_packages:
     needs: [setup, build_linux_packages]
     name: Test Linux Packages
+    strategy:
+      matrix:
+        families: ${{ fromJSON(needs.setup.outputs.linux_amdgpu_families) }}
     uses: ./.github/workflows/test_linux_packages.yml
     with:
-      amdgpu_families: "${{ inputs.amdgpu_families != '' && inputs.amdgpu_families || 'gfx94X-dcgpu' }}"
+      amdgpu_families: ${{ matrix.families.target }}
+      runs_on: ${{ matrix.families.runs-on }}
```

Comment:
Let's split these variables out by job type, e.g. `tests-run-on`. We don't want to build on test runners, but the generic `runs-on` could be inserted into the build workflows if we aren't careful during reviews.